### ISSUE TITLE
 Clear scrollback buffer as well

### ISSUE THREAD

**Issue by TG-KK** [eradman/entr#64]
Since entr uses `/usr/bin/clear` it doesn't really clear scrollback buffer.

It'd be great if entr could clear it using existing `-c` flag or maybe a new flag `-b` to clear the scrollback buffer as well, using either of these techniques:

https://apple.stackexchange.com/questions/31872/how-do-i-reset-the-scrollback-in-the-terminal-via-a-shell-command

**Comment by eradman** [comment]
I don't think this is a feature I would add to `entr`.  As far as I know there isn't a portable way to clear the scollback buffer.

You could try rebuilding with a slight change to the source: replace `/usr/bin/clear` with `/usr/bin/reset`. (On some platforms reset does a lot more than just clear the screen and is slow to respond.)

**Comment by TG-KK** [comment]
Right now I'm using `printf '\e[2J\e[3J\e[H'` to clear it without any issues, as it uses ANSI escape codes, which fish, zsh & recent versions of bash would have no issues in recognizing it.
Basically any shell that allows ANSI codes would make it work.

It's ok to not include it. Anyone else looking for it, can use something like this, to achieve the same:

```fish
function build --description 'Rerun script on changes'
  export dt=(date)
  ls $argv | entr -nr bash -c "printf '\e[2J\e[3J\e[H' && echo -e \"\n\n-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x  `date`  x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x-x\n\n\" && ./$argv"
end
```

**Comment by eradman** [comment]
The escape codes you listed may be more portable than I thought. It did work everywhere I tried

* st/xfce4-terminal on OpenBSD desktop
* Serial console to an OpenBSD router
* gnome-terminal on CentOS 7 desktop
* VM console on an x86_64 machine running Linux

I'll keep this ticket open in case anyone else whats to chime in

**Comment by eradman** [comment]
Part of the reason I thought this was complicated is because of historic nonsense

```
$ wc -l /etc/termcap
   22356 /etc/termcap
```

`clear` on OpenBSD is a link to tput(1), which reads this huge file for terminal-specific instructions

Since this seems reliable I implemented a this feature in commit 920689e which allows you to specify `-c` twice to reset the scrollback buffer.

**Comment by eradman** [comment]
This feature will be included in the next release. Closing!
